database/migrations/1749227968774_alter_field_of_studies_and_add_study_types_table.ts

```
@@ -0,0 +1,71 @@
+import { BaseSchema } from "@adonisjs/lucid/schema";
+
+export default class extends BaseSchema {
+  protected tableName = "fields_of_studies";
+  protected studiesTypes = ["1DEGREE", "2DEGREE", "LONG_CYCLE"];
+
+  async up() {
+    await this.schema.raw(`
+      DO $$ BEGIN
+        IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'studies_type') THEN
+          CREATE TYPE "studies_type" AS ENUM (${this.studiesTypes.map((type) => `'${type}'`).join(", ")});
+        END IF;
+      END $$;
+      DO $$ BEGIN
+        IF NOT EXISTS (
+            SELECT 1
+            FROM information_schema.columns
+            WHERE table_name = '${this.tableName}' AND column_name = 'studies_type'
+        ) THEN ALTER TABLE "${this.tableName}" ADD COLUMN "studies_type" "studies_type" NULL;
+        END IF;
```

no, seriously - why are you trying to detect if a column already exists
it won't, we know the migrations that were created before this one
and if it does, then i'd rather it throw an error in CI instead of being silently ignored

---

Same as above